tpl/collections/reflect_helpers.go

```diff
@@ -14,26 +14,96 @@
 package collections
 
 import (
+	"errors"
+	"fmt"
 	"reflect"
+	"time"
 )
 
-func numberToFloat(v reflect.Value) float64 {
+var (
+	zero      reflect.Value
+	errorType = reflect.TypeOf((*error)(nil)).Elem()
+	timeType  = reflect.TypeOf((*time.Time)(nil)).Elem()
+)
+
+func numberToFloat(v reflect.Value) (float64, error) {
 	switch kind := v.Kind(); {
 	case isFloat(kind):
-		return v.Float()
+		return v.Float(), nil
 	case isInt(kind):
-		return float64(v.Int())
-	case isUInt(kind):
-		return float64(v.Uint())
+		return float64(v.Int()), nil
+	case isUint(kind):
+		return float64(v.Uint()), nil
 	case kind == reflect.Interface:
 		return numberToFloat(v.Elem())
 	default:
-		panic("Invalid type in numberToFloat")
+		return 0, fmt.Errorf("Invalid kind %s in numberToFloat", kind)
+	}
+}
+
+// There are potential overflows in this function, but the downconversion of
+// int64 etc. into int8 etc. is coming from the syntetic unit tests for Union etc.
```

Comment:
Should be "synthetic", i.e. not natural -- I don't think you'll find `int8` in any real use.